ui/src/index.js

```diff
@@ -24,11 +26,37 @@ if (window.Cypress) window.__store__ = store;
 
 sagaMiddleware.run(sagas);
 
+const RouterWithBaseName = ({children}) => {
+  const configStatus = useTypedSelector(state => state.config.status);
+  const basename = useTypedSelector(state => state.config.api?.ui_base_path);
+  const [history, setHistory] = useState(createBrowserHistory({}));
+  const dispatch = useDispatch();
+  useLayoutEffect(() =>{
+    if (basename) {
+      const historyWithBasename = createBrowserHistory({basename});
+      setHistory(historyWithBasename);
+      if (window.Cypress) window.__history__ = historyWithBasename;
+      dispatch(setReduxHistory(historyWithBasename));
+    }
+    if (window.Cypress) window.__history__ = history;
+  }, [basename]);
+ 
+  if (configStatus === 'error') {
+    return <>An error occurred, please try to refresh the page</>//Todo display an error page
+  }
+
+  if (configStatus === 'idle' || configStatus === 'loading') {
+    return <>{children}</>
+  }
```

Comment:
What is this behaviour meaning? All `<Route/>`s will be rendered at once?

Reply:
Not exactly the `<App />` component is the one responsible for fetching the configuration so it needs to be rendered firstly. However we can't render the router until the configuration is not fetched because we not yet have the base path. So we render children here, to render App, which triggers config fetching when the status is idle and then when it is successfully fetched we wrap the App around the router.